General clean-up and usability for beginners

Tetragramm opened the pull request:
Don't know if you care about this still, but here's some improvements I made.

Remove flashing screen.
Can exit by pressing c.
Pauses when Karel tries to make an illegal move. Any key but 'c' continues, 'c' exits.
Modify calls to be simple functions like move(), turn_left(), ect.  No need to call execute(blah)
Add Speed control.  Can run fast or slow.  0 is slow, 100 is fast.

Tetragramm committed:
Remove flashing screen.
Can exit by pressing c.
Pauses when Karel tries to make an illegal move.
Modify calls to be simple functions like move(), turn_left(), ect.  No need to call execute(blah)
Add Speed control.  Can run fast or slow.  0 is slow, 100 is fast.

alts commented:
Ah! This is great! Thank you so much.

Could you share what python version and what platform you're using? I can't get your changes to work, and I think that's because a couple of them (for example `char = chr(char)`) are Windows specific. 

Could you also do me a favor and try running the following script on `levels\02_potholes.karelmap` against the master branch, and see if it still flashes on your screen?

```python
#/usr/bin/env python
from runner import *

def fill_pothole():
    turn_right()
    move()
    put_beeper()
    turn_around()
    move()
    turn_right()

def turn_right():
    turn_left()
    turn_left()
    turn_left()

def turn_around():
    turn_left()
    turn_left()

@execute
def run():
    move()
    fill_pothole()
    move()
```

It turns out I forgot to commit my example script, and I wonder if the flashing persists if you use `execute` only as a decorator.

I love the speed setting, and really appreciate the expansion of the README alongside the example program 🙏  I'm really excited to get this merged once we figure out some of the platform differences

Tetragramm commented:
I think some or all of this is python2->python3.  I'm using python 3.
That said, no flashing when I execute that script on master (After fixing the SyntaxErrors and LogicErrors)

I see what you are doing now, and it's obvious all the stuff I did with the move, turn_left ect was unnecessary.

Tetragramm commented:
Well, it may be unnecessary, but I think I would still keep it.  I'm not sure exactly how things are being executed.  I wouldn't be the one to try and explain decorators and just exactly how the run function is being called to total newbie programmers.

alts commented on runner.py:
```diff
@@ -6,7 +6,7 @@
     _board = Board(sys.argv[1])
 else:
-    _board = Board('simple.karelmap')
+    _board = Board('levels\\09_newspaper.karelmap')
```
This path doesn't work on my machine. I think we can replace it with the following and it should also work on your's.

```suggestion
    _board = Board(os.path.join('levels', '09_newspaper.karelmap'))
```

alts commented on board.py:
```diff
@@ -9,6 +9,7 @@ class LogicException(Exception):
 class Board(object):
     KAREL_CHARS = '<^>v'
     BEEPER_CHAR = 'o'
+    speed = 50
```
love this change. thank you 👏

alts commented on runner.py:
```diff
@@ -15,28 +15,29 @@ def inner():
+    except Exception as e:
         raise e
-    finally:
-        _board.end_screen()
```
This line here was the main reason for adding `execute`, and for using it as a decorator on some sort of `main` function. Without calling `curses.endwin()`, my bash session gets left in a bad state. I looked into some ways to get that same functionality without needing to introduce the language learner to decorators (since you're right that it's preferable to not have to add one more new unexplained thing).

One approach is to add the following after we create `_board` in `runner.py`.
```python
@atexit.register
def end_curses():
    _board.end_screen()
```

alts commented on runner.py:
```diff
@@ -15,28 +15,29 @@ def inner():
 def execute(callback):
     try:
         _board.start_screen()
         _board.draw()
```
Since `execute` gets called multiple times now, I think we should lift these two lines, the calls to `.start_screen` and `.draw`, to right after we instantiate `_board`. We already call `.draw` in `_refresh`, and `.start_screen` should only get invoked once near the beginning of running.

alts reviewed (changes requested):
Thank you so much for doing this!

I've made a few suggestions that get this change working cleanly on my machine. 

One thing that is lost in this change is that once the learner's script is finished, the curses window closes right away. When I was teaching with this some years ago, being able to visually inspect the end state was really helpful. That said, I don't have strong enough feelings on that, and I'd rather let your change land to get this functioning in Python 3 first.

Thank you again 🙏

Tetragramm committed:
Cross-platform path.
Initialize screen in constructor.
Use destructor to make sure curses ends screen.
Move exception handling into refresh, get rid of returning the callback, since we don't need it.
Add completion message, and wait for key to exit.

Tetragramm commented:
Ok, I included several of the changes, and made a few alterations.  I moved the screen init into the constructor, and the close into the destructor of the Board class.  That way, it doesn't depend on runner.py being involved at all.  runner.py still adds convenience, but it's not necessary for correct function.

Since we don't need execute at all, since we aren't decorating, I just made it so the refresh function actually does what the name implies and executes the function and re-draws the board.  Put the exception handling in there.

The pathing looks good, and works on my machine.

Printed a message in green once the program is complete and wait for input so students can view the final state.

alts commented:
👏 works for me locally. Thank you, and merged!
